src/main/java/liuni/TwitterTimeline.java

```diff
@@ -1,30 +1,31 @@
 package liuni;
 
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
 import twitter4j.Twitter;
 import twitter4j.TwitterFactory;
-import twitter4j.TwitterException;
 import twitter4j.Status;
+import twitter4j.TwitterObjectFactory;
+import twitter4j.conf.ConfigurationBuilder;
 
 import java.util.List;
 
 public class TwitterTimeline {
-    public void getTimeline() {
-        Twitter twitter = TwitterFactory.getSingleton();
-        try {
-            List<Status> statuses = twitter.getHomeTimeline();
-            System.out.println("Fetching your home timeline...");
-            for (Status status : statuses) {
-                System.out.println(status.getUser().getName() + ": " + status.getText());
-                System.out.println(" MORE INFO =======================================");
-                System.out.println(status.toString());
-                System.out.println("==================================================");
+    public JSONObject[] getTimelineJson() throws Exception {
+        ConfigurationBuilder cb = new ConfigurationBuilder();
+        cb.setJSONStoreEnabled(true);
+        Twitter twitter = new TwitterFactory(cb.build()).getInstance();
 
-            }
-        }
-        catch (TwitterException e) {
-            System.out.println("An error occurred in fetching your timeline!");
-            e.printStackTrace();
-            System.exit(-1);
+        List<Status> statuses = twitter.getHomeTimeline();
+
+        JSONObject[] jsonObjects = new JSONObject[statuses.size()];
+        JSONParser parser = new JSONParser();
```

Comment:
thoughts on getting rawJson and then successively parsing it from a performance standpoint?

Reply:
the getRawJSON method does not return a JSON object, it returns a string (any other method of getting a JSON object through the twitter4j library returns a Twitter JSON object which is more difficult to use). Parsing it allows me to convert it to a JSON object I can use more easily.